parlai/scripts/train_model.py

```diff
@@ -87,6 +87,18 @@ def setup_args(parser=None) -> ParlaiParser:
         '--evaltask',
         help='task to use for valid/test (defaults to the one used for training)',
     )
+    train.add_argument(
+        '--final-extra-valid-opt-filepath',
+        type=str,
+        default='',
+        help="A '.opt' file that is used for final eval. Useful for setting skip-generation to false. 'datatype' must be included as part of the opt.",
+    )
+    train.add_argument(
+        '--write-log-as-json',
```

Comment:
Aren't the final evals stored in the trainstats file already? If not, my suggestion would be we include them in the trainstats.

Reply:
And not even make it an option. Just do it.